packages/core/src/routes/experience/classes/experience-interaction.ts

```diff
@@ -342,6 +356,34 @@ export default class ExperienceInteraction {
     );
   }
 
+  /**
+   * Verify the captcha token using current tenant's captcha provider.
+   *
+   * @param token The captcha token to verify.
+   *
+   * @throws {RequestError} with 422 if the captcha verification fails
+   */
+  public async verifyCaptcha(token: string) {
+    const captchaProvider = await this.tenant.queries.captchaProviders.findCaptchaProvider();
```

Comment:
Will this introduce an additional query step for all SIE requests? Should we consider adding a Redis cache to optimize it?

Reply:
LOG-11057